src/main/java/uk/gov/digital/ho/pttg/application/HmrcClient.java

```diff
@@ -117,4 +126,9 @@ private void storeSelfAssessmentResource(String accessToken, String fromTaxYear,
             context.selfAssessmentResource(hateoasClient.getSelfAssessmentResource(accessToken, fromTaxYear, toTaxYear, context.getIncomeLink(SELF_ASSESSMENT)));
         }
     }
+
+    private String earliestAllowedTaxYear() {
+        return getTaxYear(LocalDate.now(clock).minusYears(maximumTaxYearHistory));
```

Comment:
What's the clock for?

Reply:
It was to assist testing. I have managed to test without it so I've now removed it.